cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -602,52 +828,155 @@ class AsofJoinNode : public ExecNode {
 
  public:
   AsofJoinNode(ExecPlan* plan, NodeVector inputs, std::vector<std::string> input_labels,
-               const AsofJoinNodeOptions& join_options,
+               const vec_col_index_t& indices_of_on_key,
+               const std::vector<vec_col_index_t>& indices_of_by_key, OnType tolerance,
                std::shared_ptr<Schema> output_schema);
 
+  Status InternalInit(bool must_hash, bool nullable_by_key,
+                      std::vector<std::unique_ptr<KeyHasher>> key_hashers) {
+    key_hashers_.swap(key_hashers);
+    auto inputs = this->inputs();
+    for (size_t i = 0; i < inputs.size(); ++i) {
+      state_.push_back(::arrow::internal::make_unique<InputState>(
+          must_hash, nullable_by_key, key_hashers_[i].get(), inputs[i]->output_schema(),
+          indices_of_on_key_[i], indices_of_by_key_[i]));
+    }
+
+    col_index_t dst_offset = 0;
+    for (auto& state : state_)
+      dst_offset = state->InitSrcToDstMapping(dst_offset, !!dst_offset);
+
+    return Status::OK();
```

Comment:
When would this return Status other than "OK"?

Reply:
This is a leftover from an older version of my code. I'll simplify to a `void` method.